docs/source/llm-eval/llm-eval.rst

```diff
@@ -0,0 +1,470 @@
+                "through its components like Spark SQL for structured data, Spark Streaming for "
+                "real-time data processing, and MLlib for machine learning tasks",
+            ],
+        }
+    )
+
+    with mlflow.start_run() as run:
+        system_prompt = "Answer the following question in two sentences"
+        # Wrap "gpt-3.5-turbo" as an MLflow model.
+        logged_model_info = mlflow.openai.log_model(
+            model="gpt-3.5-turbo",
+            task=openai.ChatCompletion,
+            artifact_path="model",
+            messages=[
+                {"role": "system", "content": system_prompt},
+                {"role": "user", "content": "{question}"},
+            ],
+        )
+
+        # Use predefined question-answering metrics to evaluate our model.
+        results = mlflow.evaluate(
+            logged_model_info.model_uri,
+            eval_data,
+            targets="ground_truth",
+            model_type="question-answering",
+        )
+        print(f"See aggregated evaluation results below: \n{results.metrics}")
+
+        # Evaluation result for each data record is available in `results.tables`.
+        eval_table = results.tables["eval_results_table"]
+        print(f"See evaluation table below: \n{eval_table}")
+
+
+LLM Evaluation Metrics
+=======================
+
+There are two types of LLM evaluation metrics in MLflow:
+
+1. Metrics relying on SaaS model (e.g., OpenAI) with prompt engineering, e.g., :py:func:`mlflow.metrics.relevance`, as called 
+   LLM-as-judge metrics. These  metrics are created via :py:func:`make_genai_metric` method. For each data record, these metrics 
+   under the hood sends one prompt consisting of the following information to the SaaS model, and extract the score from model response:
+
+   * Metrics definition.
+   * Metrics grading criteria.
+   * Reference examples.
+   * Input data/context.
+   * Model output.
+   * [optional] Ground truth.
+
+   More details of how these fields are set can be found in the section "Create your Custom LLM-evaluation Metrics".
+
+2. Function-based per-row metrics. These metrics calculate a score for each data record (row in terms of Pandas/Spark dataframe),
+   based on certain functions, like Rouge (:py:func:`mlflow.metrics.rougeL`) or Flesch Kincaid (:py:func:`mlflow.metrics.flesch_kincaid_grade_level`). 
+   These metrics are similar to traditional metrics.
+
+
+Select Metrics to Evaluate
+--------------------------
+
+There are two ways to select metrics to evaluate your LLMs:
+
+1. Use ``model_type`` argument in ``mlflow.evaluate``, every predefined model type comes with a 
+   set of metrics. This is suitable if your model falls in our predefined categories, e.g., 
+   ``question-answering``. You can find the full list of model types, and the associated evaluate 
+   metrics [here](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate). A sample code is as below:
+
+    .. code-block:: python
```

Comment:
Please wrap all code-block entries with a 

```text
.. code-section::
    .. code-block:: python
```
so that we get an indication of language

Reply:
Thanks! I will do it as a followup since without `.. code-section::` gives a better rendering in preview, which makes code reviewer easy.